src/injected-command-handler.ts

```diff
@@ -0,0 +1,456 @@
+      }
+      this.clearOverlays();
+    }
+  }
+
+  async COMMAND_TYPE_CLICKABLE(data: any): Promise<any> {
+    let response: any = {
+      clickable: false,
+    };
+    if (this.overlays.length === 0) {
+      return response;
+    }
+    const pathNumber = parseInt(data.path, 10);
+    if (!isNaN(pathNumber)) {
+      // if path is a number, check that it is available
+      response.clickable =
+        pathNumber - 1 >= 0 && pathNumber - 1 < this.overlays.length;
+    } else {
+      // otherwise, search for matching nodes
+      let matches = this.nodesMatchingPath(data.path);
+      if (matches.length >= 1) {
+        response.clickable = true;
+      }
+    }
+    return response;
+  }
+
+  async COMMAND_TYPE_DIFF(data: any): Promise<any> {
+    const editor = await editors.active();
+    if (!editor) {
+      return;
+    }
+    await editor.setSourceAndCursor(data.source, data.cursor);
+  }
+
+  async COMMAND_TYPE_DOM_BLUR(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element !== null) {
+      element.blur();
+    }
+  }
+
+  async COMMAND_TYPE_DOM_CLICK(data: any): Promise<any> {
+    const node = document.querySelector(data.text);
+    if (node) {
+      this.clickNode(node);
+    }
+  }
+
+  async COMMAND_TYPE_DOM_COPY(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element && window.getSelection) {
+      let selection = window.getSelection();
+      let range = document.createRange();
+      range.selectNodeContents(element);
+      selection?.removeAllRanges();
+      selection?.addRange(range);
+      await navigator.clipboard.writeText(selection?.toString() || "");
+      selection?.removeAllRanges();
+    }
+  }
+
+  async COMMAND_TYPE_DOM_FOCUS(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element) {
+      element.focus();
+    }
+  }
+
+  async COMMAND_TYPE_DOM_SCROLL(data: any): Promise<any> {
+    const element = document.querySelector(data.text) as HTMLElement;
+    if (element) {
+      element.scrollIntoView({
+        block: "center",
+        inline: "center",
+        behavior: "smooth",
+      });
+    }
+  }
+
+  async COMMAND_TYPE_GET_EDITOR_STATE(_data: any): Promise<any> {
+    const editor = await editors.active();
+    if (!editor) {
+      return;
+    }
+    return editor.getEditorState();
+  }
+
+  async COMMAND_TYPE_REDO(_data: any): Promise<any> {
+    const editor = await editors.active();
+    editor?.redo();
+    return;
+  }
+
+  async COMMAND_TYPE_SCROLL(data: any): Promise<any> {
+    if (data.direction) {
+      this.scrollInDirection(data.direction);
+    }
+    if (data.path) {
+      if (data.path === "top" || data.path === "bottom") {
+        return this.scrollInDirection(data.path);
+      } else {
+        return this.findAndScroll(data.path);
+      }
```

Comment:
I think we only need one level of nesting here—either `data.path` equals `top` or `bottom`, or it's not falsey and we want to call `findAndScroll`.